openedx/core/lib/block_structure/transformer.py

```diff
@@ -125,4 +126,55 @@ def transform(self, usage_info, block_structure):
                 block structure, with already collected data for the
                 transformer, that is to be transformed in place.
         """
-        pass
+        raise NotImplementedError
+
+
+class OptimizedTransformer(BlockStructureTransformer):
```

Comment:
Some transformers use PostOrder-based traversal, while others use PreOrder.  It depends on their requirements.  Some may also just find all blocks with a filter - without traversing.  Depending on what's faster.  My overall point being, that all Transformer development should be implemented/reviewed with performance and optimization always in mind.